### PR TITLE
Bug/gh331 error opening citation dialog; fixes #331

### LINKED ISSUE #331: Error opening Citation dialog
- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4-IME
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Citation dialog doesn't show because of an error.

Problem occurs in method PopulatePossibleSource in CDP4CommonView.ViewModels.CitationViewModel:

```
        protected override void PopulatePossibleSource()
        {
            this.PossibleSource.Clear();
            IEnumerable<ReferenceSource> referenceSources = null;
            var rdlsInChain = this.ChainOfContainer.Where(x => x is ReferenceDataLibrary).ToList();
            if (rdlsInChain.Any())
            {
                referenceSources = rdlsInChain.SelectMany(x => ((ReferenceDataLibrary)x).ReferenceSource);
            }

            this.PossibleSource.AddRange(referenceSources);
        }
```

Somehow, the ChainOfContainer property should contain a ReferenceDataLibrary.
In my local database that's not the case, so referenceSources variable is null.
That lets the AddRange method crash.

The crash should at least not be happening, but the reason that the ReferenceDataLibrary is not in ChainOfContainer (should it even be in there?) also has to be investigated.

### Steps to Reproduce
- Start a edit dialog that has a Definition tab (for example PossibleFiniteState edit dialog)
- Go to the Definitions tab
- Press the add button to add a Definition
- Select the Citation tab
- Press the add button to add a citation
- No Citation dialog is opened en the status bar shows that an error occurred

If the Citation dialog does show up, then I have a data issue locally, but then the PopulatePossibleSource method imho still needs some refactoring.

- CDP4 version:
  - [x] CDP4-IME: 5.1.1
  - [ ] Other:

### PR DESCRIPTION
**Pull request by kamilatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

- I added constraint that validate if chain of container not contain ReferenceDataLibrary, then cannot be add to possible source of CitationDialog
- UI create button react on that and disable the add button

### CODE REVIEW

File: CDP4CommonView/ViewModels/CitationDialogViewModel.cs [hunk 1]
```diff
@@ -94,11 +94,13 @@ protected override void PopulatePossibleSource()
             var rdlsInChain = this.ChainOfContainer.Where(x => x is ReferenceDataLibrary).ToList();
             if (rdlsInChain.Any())
             {
-                referenceSources = rdlsInChain.SelectMany(x => ((ReferenceDataLibrary)x).ReferenceSource);
+                referenceSources = rdlsInChain.SelectMany(x => ((ReferenceDataLibrary)x).ReferenceSource).OrderBy(x=>x.Name);
```
Comment (alexatrhea) [inline CDP4CommonView/ViewModels/CitationDialogViewModel.cs:97]: Spacing around `=>`

Comment (kamilatrhea) [reply CDP4CommonView/ViewModels/CitationDialogViewModel.cs]: Ok

File: CDP4CommonView/AutoGenDialogViewModel/DefinitionDialogViewModel.cs [hunk 2]
```diff
@@ -1,6 +1,25 @@
 ﻿// -------------------------------------------------------------------------------------------------
 // <copyright file="DefinitionDialogViewModel.cs" company="RHEA System S.A.">
 //   Copyright (c) 2015-2017 RHEA S.A.
```
Comment (lxatrhea) [inline CDP4CommonView/AutoGenDialogViewModel/DefinitionDialogViewModel.cs:3]: please change to 2015-2020

Comment (kamilatrhea) [reply CDP4CommonView/AutoGenDialogViewModel/DefinitionDialogViewModel.cs]: Ok

**Pull request merged**